src/github.com/getlantern/autoupdate/autoupdate.go

```diff
@@ -0,0 +1,169 @@
+// Package autoupdate provides Lantern with special tools to autoupdate itself
+// with minimal effort.
+package autoupdate
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/getlantern/flashlight/util"
+	"github.com/getlantern/go-update"
+	"github.com/getlantern/go-update/check"
+)
+
+const noVersion = -1
+
+// Making sure AutoUpdate and Patch satisfy AutoUpdater and Patcher.
+var (
+	_ = AutoUpdater(&AutoUpdate{})
+	_ = Patcher(&Patch{})
+)
+
+var (
+	// How much time should we wait between update attempts?
+	sleepTime = time.Hour * 4
+)
+
+// SetProxy sets the proxy to use.
+func SetProxy(proxyAddr string) {
+	var err error
+
+	if proxyAddr != "" {
+		// Create a HTTP proxy and pass it to the update package.
+		if update.HTTPClient, err = util.HTTPClient("", proxyAddr); err != nil {
+			log.Printf("Could not use proxy: %q\n", err)
+		}
+	} else {
+		update.HTTPClient = &http.Client{}
+	}
+}
+
+// AutoUpdate satisfies AutoUpdater and can be used for other programs to
+// configure automatic updates.
+type AutoUpdate struct {
+	cfg config
+	v   int
+	// When a patch has been applied, the patch's version will be sent to
+	// UpdatedTo.
+	UpdatedTo chan int
+}
+
+// New creates an AutoUpdate struct based in the configuration defined in
+// config.go.
+func New(appName string) *AutoUpdate {
+	if configMap[appName] == nil {
```

Comment:
Thats right. The original reasoning behind having the autoupdates configuration within the `autoupdate` package and outside the application that is going to use it, is to reduce the amount of code that depends on the structure of `autoupdate` given that it could change, for example, if we ever decide changing our backend and not using Equinox.IO that configuration struct would surely change.

I think this is the best way to keep packages completely independent from Equinox.IO, since the configuration struct is modelled after the Equinox.IO platform's needs. In fact, packages that import `autoupdate` do not need to know that they're going to communicate with Equinox.IO at all.

At this time we're targeting only one application, the Lantern binary, so changing the configuration struct would not be that hard if we decide to move the configuration outside the `autoupdate` package.